src/librustc_typeck/check/cast.rs

```diff
@@ -346,9 +350,11 @@ impl<'a, 'gcx, 'tcx> CastCheck<'tcx> {
             ty::Adt(def, ..) if def.is_box() => {
                 match fcx.tcx.sess.source_map().span_to_snippet(self.cast_span) {
                     Ok(s) => {
-                        err.span_suggestion(self.cast_span,
+                        err.span_suggestion_with_applicability(self.cast_span,
                                             "try casting to a `Box` instead",
-                                            format!("Box<{}>", s));
+                                            format!("Box<{}>", s),
+                                            Applicability::MachineApplicable,
```

Comment:
Side question: If I read this correctly, `s` is a trait here. Should we suggest `Box<dyn {}>` here? (That's how you use the `dyn` keyword, right?) Probably the same for "try casting to a reference …" above